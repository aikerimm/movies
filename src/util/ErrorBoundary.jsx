import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { getMoviesData } from './moviesSlice.jsx';

const ErrorBoundary = (props) => {
  const movies = useSelector(getMoviesData);
  const FallbackText = () => <h2>Oops, no movies were found.</h2>;

  let isOk = movies.data.length > 0;
  return <>{isOk ? props.children : <FallbackText />}</>;
};

ErrorBoundary.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default React.memo(ErrorBoundary);
