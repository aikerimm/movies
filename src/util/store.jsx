import { configureStore } from '@reduxjs/toolkit';
import moviesReducer from './moviesSlice.jsx';

export default configureStore({
  reducer: {
    movies: moviesReducer
  }
});