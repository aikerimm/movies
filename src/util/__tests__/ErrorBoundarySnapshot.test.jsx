import renderer from 'react-test-renderer';
import ErrorBoundary from '../ErrorBoundary';
import { Provider } from 'react-redux';
import store from '../store.jsx';

it('ErrorBoundary should render correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <ErrorBoundary>
          <h1>Success</h1>
        </ErrorBoundary>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
