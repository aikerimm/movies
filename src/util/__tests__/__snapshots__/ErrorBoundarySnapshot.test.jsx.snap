// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should render correctly 1`] = `
<h2>
  Oops, no movies were found.
</h2>
`;
